pages/home.py

```diff
@@ -106,20 +113,49 @@ def update_store_inputs(
     inputs[ElementsIDs.MODEL_SELECTION.value] = selected_model
     inputs[ElementsIDs.chart_selected.value] = chart_selected
 
-    return inputs
+    print(f"inputs: {inputs}")
+    url_data = inputs
+    # encode the inputs to be used in the URL
+    url_search = f"?{urlencode(inputs)}"
+    print(f"url_params: {url_data}")
+
+    return inputs, url_search
 
 
 @callback(
     Output(ElementsIDs.INPUT_SECTION.value, "children"),
     Input(ElementsIDs.MODEL_SELECTION.value, "value"),
     Input(ElementsIDs.UNIT_TOGGLE.value, "checked"),
+    Input("url", "search"),
```

Comment:
do not use a string "url" but instead create an attribute to ElementsIDs called url

Reply:
fit it

Reply:
I will fix it